core/domain/suggestion_services.py

```diff
@@ -618,6 +618,9 @@ def get_reviewable_translation_suggestions_by_offset(
         user_id)
     language_codes = (
         contribution_rights.can_review_translation_for_language_codes)
+    # The user cannot review any translations, so return early.
+    if len(language_codes) == 0:
+        return [], 0
```

Comment:
Should the second element be None to indicate "no further results", rather than "start again from 0 on the next query"?

Reply:
I think returning the input offset makes sense to fulfill the method contract: "The input offset + the number of results returned by the current query." Updated.

Reply:
But then how do you signal to the caller when there's no more results to fetch?

Reply:
Well if there are no more results to fetch, then both the results list should be empty and the returned offset shouldn't change. The caller could use either as a signal.

Reply:
But what about the case where the results list isn't empty and there are no more results to fetch? E.g. let's say there are 100 results, my offset is 80, I ask for 20, the backend provides 20 results (and there are no more to fetch). Shouldn't there be some way to distinguish this from the other case of "there are 101 results" so that the client knows whether to show a "get more" button or an "end of results" button?

Reply:
Ah, gotcha. I believe we've decided to give the responsibility of computing whether there are more results to the frontend client (https://github.com/oppia/oppia/issues/13462) rather than implicitly fetching ahead in the backend.

Reply:
The `next_offset` return field is a bit redundant, but it does save the caller from having to compute `next_offset = offset + len(results)` after each call. 

The filtering doesn't occur in this method. In this PR we're also not using the offset at all.

Reply:
Yeah, I guess that's true (re the filtering).

OK, thanks -- lgtm!